src/Orc.FileAssociation/Services/FileAssociationService.cs

```diff
@@ -7,44 +7,56 @@
 
 namespace Orc.FileAssociation
 {
+    using System;
     using System.Runtime.InteropServices;
     using Catel;
     using Catel.Logging;
+    using Microsoft.Win32;
 
     public class FileAssociationService : IFileAssociationService
     {
         private static readonly ILog Log = LogManager.GetCurrentClassLogger();
 
-        public void AssociateFilesWithApplication(string applicationName = null)
+        public void AssociateFilesWithApplication(ApplicationInfo applicationInfo)
```

Comment:
This is a breaking change and will break apps using this API. Please introduce a new overload with this parameter.